qiskit/pulse/filters.py

```diff
@@ -107,17 +174,39 @@ def with_channels(channels: Union[Iterable[Channel], Channel]) -> Callable:
     """
     channels = _if_scalar_cast_to_list(channels)
 
-    def channel_filter(time_inst) -> bool:
+    @singledispatch
+    def channel_filter(time_inst):
+        """A catch-TypeError function which will only get called if none of the other decorated
+        functions, namely handle_numpyndarray() and handle_instruction(), handle the type passed.
+        """
+        raise TypeError(
+            f"Type '{type(time_inst)}' is not valid data format as an input to channel_filter."
+        )
+
+    @channel_filter.register
+    def handle_numpyndarray(time_inst: np.ndarray) -> bool:
         """Filter channel.
 
         Args:
-            time_inst (Tuple[int, Instruction]): Time
+            time_inst (numpy.ndarray([int, Instruction])): Time
```

Comment:
I noticed that the type of objects passed to this function is in fact not `tuple` but `numpy.ndarray`, as I updated here.
The reason is because the following code breaks `tuple` and makes it to `numpy.ndarray`:
https://github.com/Qiskit/qiskit-terra/blob/bedecbdce563f4e83acc11c7ec5ca6878a36a4b7/qiskit/pulse/filters.py#L48